Multiple mapped tiles when creating synth io tiles

Added a synth tile definition to the design.json as follows:
`        {
            "name": "dout[8]",
            "node": "INT_R_X23Y100/LH3",
            "pin": "H7",
            "wire": "INT_INTERFACE_R_X23Y100/INT_INTERFACE_LH3",
            "wires_outside_roi": [
            ]
        }
`

And building the synth tiles gives the following error:
`  File "/scratch/safe/butta/symbiflow-arch-defs.syn/xc/common/utils/prjxray_create_synth_tiles.py", line 26, in map_tile_to_vpr_coord
    assert len(mapped_tiles) == 1, tile
AssertionError: INT_INTERFACE_R_X23Y100
`
I know this is not a normal location used as a synth tile, but is this a location I should be able to place a synth tile?  Is mapping to multiple vpr coordinates the correct result in this situation or a bug?

Link to branch containing these changes: [synth-tile-errors](https://github.com/andrewb1999/symbiflow-arch-defs/tree/synth-tile-errors)

--- a/xc/common/utils/prjxray_create_synth_tiles.py
+++ b/xc/common/utils/prjxray_create_synth_tiles.py
@@ -16,11 +16,14 @@ def map_tile_to_vpr_coord(conn, tile):
     c.execute("SELECT pkey FROM phy_tile WHERE name = ?;", (tile, ))
     phy_tile_pkey = c.fetchone()[0]
 
+    c.execute("SELECT pkey FROM tile_type WHERE name = 'NULL'")
+    null_tile_type_pkey, = c.fetchone()
+
     # It is expected that this tile has only one logical location,
     # because why split a tile with no sites?
     c.execute(
-        "SELECT tile_pkey FROM tile_map WHERE phy_tile_pkey = ?",
-        (phy_tile_pkey, )
+        "SELECT tile_pkey FROM tile_map WHERE phy_tile_pkey = ? AND tile_type_pkey != ?",
+        (phy_tile_pkey, null_tile_type_pkey)
     )
     mapped_tiles = c.fetchall()
     assert len(mapped_tiles) == 1, tile

--- a/xc/common/utils/prjxray_create_synth_tiles.py
+++ b/xc/common/utils/prjxray_create_synth_tiles.py
@@ -23,8 +23,9 @@ def map_tile_to_vpr_coord(conn, tile):
     # because why split a tile with no sites?
     c.execute(
         """
-SELECT tile_pkey FROM tile_map INNER JOIN phy_tile ON tile_map.phy_tile_pkey = phy_tile.pkey
-WHERE phy_tile_pkey = ? AND tile_type_pkey != ?
+SELECT tile_map.tile_pkey FROM tile_map INNER JOIN phy_tile
+ON tile_map.phy_tile_pkey = phy_tile.pkey
+WHERE tile_map.phy_tile_pkey = ? AND phy_tile.tile_type_pkey != ?
         """, (phy_tile_pkey, null_tile_type_pkey)
     )
     mapped_tiles = c.fetchall()

--- a/xc/common/utils/prjxray_create_synth_tiles.py
+++ b/xc/common/utils/prjxray_create_synth_tiles.py
@@ -22,8 +22,10 @@ def map_tile_to_vpr_coord(conn, tile):
     # It is expected that this tile has only one logical location,
     # because why split a tile with no sites?
     c.execute(
-        "SELECT tile_pkey FROM tile_map WHERE phy_tile_pkey = ? AND tile_type_pkey != ?",
-        (phy_tile_pkey, null_tile_type_pkey)
+        """
+SELECT tile_pkey FROM tile_map INNER JOIN phy_tile ON tile_map.phy_tile_pkey = phy_tile.pkey
+WHERE phy_tile_pkey = ? AND tile_type_pkey != ?
+        """, (phy_tile_pkey, null_tile_type_pkey)
     )
     mapped_tiles = c.fetchall()
     assert len(mapped_tiles) == 1, tile

--- a/xc/common/utils/prjxray_create_synth_tiles.py
+++ b/xc/common/utils/prjxray_create_synth_tiles.py
@@ -23,10 +23,11 @@ def map_tile_to_vpr_coord(conn, tile):
     # because why split a tile with no sites?
     c.execute(
         """
-SELECT tile_map.tile_pkey FROM tile_map INNER JOIN phy_tile
-ON tile_map.phy_tile_pkey = phy_tile.pkey
-WHERE tile_map.phy_tile_pkey = ? AND phy_tile.tile_type_pkey != ?
-        """, (phy_tile_pkey, null_tile_type_pkey)
+SELECT tile_map.tile_pkey FROM tile_map INNER JOIN tile
+ON tile_map.tile_pkey = tile.pkey INNER JOIN tile_type
+ON tile.tile_type_pkey = tile_type.pkey
+WHERE tile_map.phy_tile_pkey = ? AND tile_type.name != 'NULL'
+    """, (phy_tile_pkey, )
     )
     mapped_tiles = c.fetchall()
     assert len(mapped_tiles) == 1, tile

